src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+   * compilation unit.
+   *
+   * @param compilationUnit compilation unit to add import to
+   * @param name the import name
+   */
+  private static void addImport(CompilationUnit compilationUnit, String name) {
+    String i = "import " + name + ";";
+    ImportDeclaration importDeclaration;
+
+    try {
+      importDeclaration = JavaParser.parseImport(i);
+    } catch (ParseException e) {
+      // Unexpected error from parsing import.
+      System.err.println("Error parsing import");
+      return;
+    }
+
+    List<ImportDeclaration> importDeclarations = compilationUnit.getImports();
+    for (ImportDeclaration im : importDeclarations) {
+      // Check if the compilation unit already includes the import.
+      if (importDeclaration.toString().equals(im.toString())) {
+        return;
+      }
+    }
+
+    // Add the import to the compilation unit's list of imports.
+    importDeclarations.add(importDeclaration);
+    compilationUnit.setImports(importDeclarations);
+  }
+
+  /** Contains the standard output, standard error, and exit status from running a process. */
+  private static class Outputs {
+    /** The standard output */
+    private String stdout;
+    /** The error output */
+    private String errout;
+
+    /** Exit value from running a process */
+    private int exitValue;
+
+    /**
+     * Create an Outputs object.
+     *
+     * @param stdout standard output
+     * @param errout error output
+     * @param exitValue exit value of process
+     */
+    private Outputs(String stdout, String errout, int exitValue) {
+      this.stdout = stdout;
+      this.errout = errout;
+      this.exitValue = exitValue;
+    }
+  }
+
+  /** Contains two {@code Output} objects for compilation and execution. */
+  private static class Results {
+    /** The standard and error output of compiling a Java file. */
+    private Outputs compOut;
+    /** The standard and error output of running a Java file. */
+    private Outputs runOut;
+
+    /**
+     * Create a Results object.
+     *
+     * @param compOut compilation results
+     * @param runOut runtime results
+     */
+    private Results(Outputs compOut, Outputs runOut) {
+      this.compOut = compOut;
+      this.runOut = runOut;
+    }
+  }
+
+  /**
+   * Calculate the length of a file, by number of lines
+   *
+   * @param filepath absolute file path to the input file
+   * @return the number of lines in the file. Negative one is returned if an exception occurs from
+   *     finding or reading the file
+   */
+  private static int getFileLength(String filepath) {
+    int lines = 0;
+    try {
+      BufferedReader reader = new BufferedReader(new FileReader(filepath));
+      while (reader.readLine() != null) {
+        lines++;
+      }
+      reader.close();
+    } catch (FileNotFoundException e) {
+      System.err.println("File length not calculated, file not found exception.");
```

Comment:
When outputting an error message about a file I/O problem, always give the name/path of the file.